utils/build-script

```diff
@@ -148,19 +131,21 @@ def validate_xcode_compatibility():
         print("note: skipping Xcode version check")
         return
 
-    version = shell.capture(
+    output = shell.capture(
         ['xcodebuild', '-version'], dry_run=False, echo=False).strip()
+    # Capture only version ex: '14.x' from full output
+    match = re.match(r"Xcode (\d+\.\d+(?:\.\d)?)", output)
+    if not match:
+        print(f"warning: unexpected xcodebuild output format: '{output}', skipping Xcode compatibility check, please report this issue", file=sys.stderr)
+        return
 
-    valid_build_numbers = tuple(x[1] for x in _SUPPORTED_XCODE_BUILDS)
-    if not version.endswith(valid_build_numbers):
-        valid_versions_string = "\n".join(
-            "{} ({})".format(*x) for x in _SUPPORTED_XCODE_BUILDS)
+    version = match.group(1)
+    minimum_version = "13.0"
+    if version < minimum_version:
```

Comment:
if you assume that all new versions are supported (which I am, although that might be invalid), is there a specific case you can think of where this won't work?

Reply:
Good examples, I agree they might not be a huge deal, but I changed it to use your example anyways.

> I recently tried building with Xcode 14.1 on macOS Ventura, and there were failures due to mismatching Swift versions, if I recall correctly. The --bootstrapping (or another) option might have helped in that situation, so I'm not sure if pending_version is needed.

Yea I don't want to make any calls on this unless we're sure atm, but it's definitely nice to be flexible in the future if needed.

> Could the Xcode version check become a warning, instead of an error?

I think, especially without a "highest supported version" check folks should see this error _much_ less moving forward, since previously it needed a manual update which clearly folks were not doing. There was also this discussion about warnings vs errors last time https://github.com/apple/swift/pull/34227#discussion_r501332326

but hopefully all of that is moot since this should be much more permissive!